Library/include/CSP/Systems/ECommerce/ECommerce.h

```diff
@@ -0,0 +1,155 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+#include "CSP/CSPCommon.h"
+#include "CSP/Common/Map.h"
+#include "CSP/Common/String.h"
+#include "CSP/Services/WebService.h"
+
+namespace csp::services
+{
+
+CSP_START_IGNORE
+template <typename T, typename U, typename V, typename W> class ApiResponseHandler;
+CSP_END_IGNORE
+
+} // namespace csp::services
+
+
+
+namespace csp::systems
+{
+/// @ingroup ECommerce System
+/// @brief Represents currency information for a product
+class CSP_API CurrencyInfo
+{
+public:
+	CurrencyInfo() = default;
+
+	/// @brief Currency Amount.
+	double Amount;
+	/// @brief Currency type
+	csp::common::String CurrencyCode;
+};
+
+/// @ingroup ECommerce System
+/// @brief Represents media information for a product
+class CSP_API ProductMediaInfo
+{
+public:
+	ProductMediaInfo() = default;
+
+	/// @brief Type of media content used.
+	csp::common::String MediaContentType;
+	/// @brief Alternative description of the media.
+	csp::common::String Alt;
+	/// @brief Url of the media.
+	csp::common::String Url;
+	/// @brief Width of the media.
+	int32_t Width;
+	/// @brief Height of the media.
+	int32_t Height;
+};
+
+/// @ingroup ECommerce System
+/// @brief Represents option for a variant
+class CSP_API VariantOptionInfo
+{
+public:
+	VariantOptionInfo() = default;
+
+	/// @brief Id of the variant option.
+	csp::common::String Name;
+	/// @brief Value of variant option.
+	csp::common::String Value;
+};
+
+/// @ingroup ECommerce System
+/// @brief Represents variant information for a product
+class CSP_API ProductVariantInfo
+{
+public:
+	ProductVariantInfo() = default;
+	/// @brief Id of the variant.
+	csp::common::String Id;
+	/// @brief Title of the variant.
+	csp::common::String Title;
+	/// @brief Title of variant.
+	csp::common::String Url;
+	/// @brief is variant available for sale.
+	bool AvailableForSale;
+	/// @brief This image for a variant
+	ProductMediaInfo Media;
+	/// @brief Additional options for variant
+	csp::common::Array<VariantOptionInfo> Options;
+	/// @brief Unit price for the variant
+	CurrencyInfo UnitPrice;
+};
+
+
+/// @ingroup ECommerce System
+/// @brief Represents a single product and the information associated with it.
+class CSP_API ProductInfo
+{
+public:
+	ProductInfo() = default;
+
+	/// @brief Id of the product.
+	csp::common::String Id;
+	/// @brief Title of the product.
+	csp::common::String Title;
+	/// @brief  Time the product was created.
+	csp::common::String CreatedAt;
+	/// @brief Description of the product.
+	csp::common::String Description;
+	/// @brief array of product variants.
+	csp::common::Array<ProductVariantInfo> Variants;
+	/// @brief array of product tags.
+	csp::common::Array<csp::common::String> Tags;
+	/// @brief This array holds  media for the product
```

Comment:
Additional whitespace between 'holds' and 'media'

Reply:
sorted